### PR TITLE
Use Pg_Bool to support multiple kinds of pg_sys::bool_

### LINKED ISSUE #2: Fails to compile on Linux due to mismatched u8
Environment:
```
$ uname -a
Linux penguin 4.14.67-07156-gc116f2c8c400 #1 SMP PREEMPT Sun Sep 9 14:20:03 PDT 2018 aarch64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 9.5 (stretch)
Release:        9.5
Codename:       stretch
$ cargo version
cargo 1.33.0-nightly (0d1f1bbea 2018-12-19)
```

Errors:
```
error[E0308]: mismatched types
  --> pg-extend/src/lib.rs:68:30
   |
68 |     let args_null: &[bool] = &func_call_info.argnull[..num_args];
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected bool, found u8
   |
   = note: expected type `&[bool]`
              found type `&[u8]`

error[E0308]: mismatched types
  --> pg-extend/src/pg_error.rs:85:12
   |
85 |         if pg_sys::errstart(errlevel, file, line, func_name, ERR_DOMAIN.as_ptr() as *const c_char) {
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected bool, found u8
```

FunctionCallArgsData is defined as:

```rust
pub struct FunctionCallInfoData { pub flinfo : * mut FmgrInfo , pub context : fmNodePtr , pub resultinfo : fmNodePtr , pub fncollation : Oid , pub isnull : bool_ , pub nargs : :: std :: os :: raw :: c_short , pub arg : [ Datum ; 100usize ] , pub argnull : [ bool_ ; 100usize ] , }
```

Where `bool_` is:

```rust
pub type bool_ = :: std :: os :: raw :: c_char ;
```
[Full gist here](https://gist.github.com/881e2086559d0ac5a4a60981cc179eb0)

### PR DESCRIPTION
**Pull request by stillinbeta**
On some systems, bool_ is a u8 instead of a boolean.
We can use traits to handle this case.

Fixes #2

### CODE REVIEW

**Comment by bluejekyll** [comment]
This looks good. Let me know if you want to attempt to tackle that one change.

Happy New Year!

**Comment by stillinbeta** [comment]
@bluejekyll which one change is that, sorry?

File: pg-extend/src/lib.rs [hunk 1]
```diff
@@ -61,11 +60,15 @@ macro_rules! pg_magic {
...
+) -> (&[pg_sys::Datum], Vec<bool>) {
     let num_args = func_call_info.nargs as usize;
 
     let args: &[pg_sys::Datum] = &func_call_info.arg[..num_args];
-    let args_null: &[bool] = &func_call_info.argnull[..num_args];
+    let args_null: Vec<bool> =
+        func_call_info.argnull[..num_args]
+        .into_iter()
+        .map(|b| pg_bool::Bool::from(*b).into())
+        .collect();
```
[the remark this answers is not in the document]
Comment (bluejekyll) [reply pg-extend/src/lib.rs]: I'll take this. There are a few changes I want to merge in.

**Comment by bluejekyll** [comment]
Thanks for the PR, i'll handle this other change when I get a chance to come back to it.

Basically, since this will be called on every invocation of the function, we don't want to allocate a new Vec each time for the bools, I think we can avoid that with the iterator idea I mentioned. I'll get to that shortly.

**Pull request merged**